src/main.cpp

```diff
@@ -3738,6 +3759,108 @@ bool CVerifyDB::VerifyDB(CCoinsView *coinsview, int nCheckLevel, int nCheckDepth
     return true;
 }
 
+bool RewindBlockIndex(const CChainParams& params)
+{
+    LOCK(cs_main);
+
+    // RewindBlockIndex is called after LoadBlockIndex, so at this point every block
+    // index will have nBranchId set either to the persisted value, or the value of
+    // its parent block. Thus we can reason here that for valid chains:
+    //
+    // - BLOCK_OPT_ACTIVATES_NU must be set only on blocks that activate upgrades.
```

Comment:
"must be set" or "is set"? The former doesn't make sense to me because I thought this comment was making an assertion about what definitely is the case.

Reply:
Hmm, good point. It's technically both: this function is itself defining / enforcing this validity rule, but the rule is derived from the fact that `nBranchId` will only ever be set if the block did in fact satisfy its consensus rules. I'll try to reword the comment.